arrow/doc.go

```diff
@@ -34,7 +34,7 @@ To build with tinygo include the noasm build tag.
 */
 package arrow
 
-const PkgVersion = "18.0.0-SNAPSHOT"
+const PkgVersion = "18.0.0"
```

Comment:
this doesn't have to change for this PR, only when releasing, right? or do we want this to be `17.0.0` and increment on the new release?
I think we should probably continue using `SNAPSHOT` and just remove the `SNAPSHOT` bit when releasing.

Reply:
> this doesn't have to change for this PR, only when releasing, right?

No.

> or do we want this to be 17.0.0 and increment on the new release?

No.

My suggestions:
* Don't use `-SNAPSHOT` suffix for unreleased version
  * I think that it's needless because we don't create a nightly (or something) package from main
  * I think that the `-SNAPSHOT` suffix name is Maven convention: https://maven.apache.org/maven-release/maven-release-plugin/usage/prepare-release.html
    * If we want to use `-XXX` suffix for unreleased version, other name such as `-dev` is better
* We don't update this just after we release a new version
  * Because we can't know the next version (`18.0.1`? `18.1.0`? `19.0.0`?) just after we release a new version
* We update this just before we release a new version, when we add a new compatible feature (minor version up) or when we add an incompatible change (major version up)

Reply:
sounds good to me to update just before releasing